### DISCUSSION TITLE
Continuous Delivery

### DISCUSSION THREAD

**Discussion by jwfuller** [nsidc/tech-radar#5] [Techniques]
We manually trigger deployments now with an approval process. Continuous delivery would be more like: Someone (ops? devs?) push a tag to a repo, and then delivery automatically occurs in response. This may not map well to the type of work we do. One process hurdle we have is that each prod change requires a record in JIRA (PCT).

**Comment by MattF-NSIDC** [comment]
I think we _are_ actually doing "continuous delivery", and fairly widely too! Many projects use automation to trigger release to software repositories (e.g. anaconda.org, dockerhub) and test environments (e.g. trigger a Jenkins machine to deploy to integration/QA) when a tag is pushed to GitHub/BB.

I think "continuous deployment"  (automatically putting the software in production) is what we're not currently doing for the reasons you stated above.

Projects that use Garrison are probably the closest we get to continuous deployment. We tag software, it's automatically released/"delivered", then we ask ops to deploy that tag via a PCT ticket, then ops tells Jenkins to deploy that tag. Jenkins uses Garrison to do the whole deployment process in a scripted way, including stuff like database migrations. While we've stopped short of fully automating those two human steps, we have minimized the manual steps.
